src/commons/utils/JavaHelper.ts

```diff
@@ -9,6 +11,33 @@ import DisplayBufferService from './DisplayBufferService';
 export async function javaRun(javaCode: string, context: Context) {
   let compiled = {};
 
+  const stderr = (type: 'TypeCheck' | 'Compile' | 'Runtime', msg: string) => {
+    context.errors.push({
+      type: type as any,
+      severity: 'Error' as any,
+      location: { start: { line: -1, column: -1 }, end: { line: -1, column: -1 } },
+      explain: () => msg,
+      elaborate: () => msg
+    });
+  };
+
+  const typeCheckResult = typeCheck(javaCode);
+  if (typeCheckResult.hasTypeErrors) {
+    const typeErrMsg = typeCheckResult.errorMsgs.join('\n');
+    stderr('TypeCheck', typeErrMsg);
+    return Promise.resolve({ status: 'error' });
+  }
+
+  try {
+    const classFile = compileFromSource(javaCode);
+    compiled = {
+      'Main.class': Buffer.from(new BinaryWriter().generateBinary(classFile)).toString('base64')
+    };
+  } catch (e) {
+    stderr('TypeCheck', e);
```

Comment:
Looks strange: Shouldn't this be runtime error instead of TypeCheck?

Reply:
Will is be ok if I changed this to `"Compile"`? Since it catches errors during compiler time.

Reply:
The `RuntimeErrors` are caught at the return of the `javaRun` method.

Reply:
Resolved.

<img width="409" alt="image" src="https://github.com/source-academy/frontend/assets/110974917/4a95eb71-1878-4e12-a85d-d7a8cbcbb691">